process/block/sovereign/incomingHeader/topicsChecker.go

```diff
@@ -13,21 +15,58 @@ func NewTopicsChecker() *topicsChecker {
 	return &topicsChecker{}
 }
 
-// CheckValidity will receive the topics and validate them
-func (tc *topicsChecker) CheckValidity(topics [][]byte) error {
+// CheckDepositTokensValidity will receive the deposit topics and validate them
+func (tc *topicsChecker) CheckDepositTokensValidity(topics [][]byte) error {
 	// TODO: Check each param validity (e.g. check that topic[0] == valid address)
 	if len(topics) < dto.MinTopicsInTransferEvent || len(topics[2:])%dto.NumTransferTopics != 0 {
-		log.Error("incomingHeaderHandler.createIncomingSCRs",
-			"error", dto.ErrInvalidNumTopicsIncomingEvent,
+		log.Error("topicsChecker.CheckDepositTokensValidity",
+			"error", dto.ErrInvalidNumTopicsInEvent,
 			"num topics", len(topics),
 			"topics", topics)
 
-		return fmt.Errorf("%w for %s; num topics = %d", dto.ErrInvalidNumTopicsIncomingEvent, "eventIDDepositIncomingTransfer", len(topics))
+		return fmt.Errorf("%w for deposit event; num topics = %d", dto.ErrInvalidNumTopicsInEvent, len(topics))
+	}
+
+	return nil
+}
+
+// CheckScCallValidity will receive the topics and transfer data, and validate them
+func (tc *topicsChecker) CheckScCallValidity(topics [][]byte, transferData *sovereign.TransferData) error {
+	// TODO: Check each param validity (e.g. check that topic[0] == valid address, valid transferData)
+	if len(topics) != dto.NumScCallTopics || transferData == nil {
+		log.Error("topicsChecker.CheckScCallValidity",
+			"error", dto.ErrInvalidNumTopicsInEvent,
+			"num topics", len(topics),
+			"topics", topics,
+			"transferData is nil", transferData == nil)
+
+		return fmt.Errorf("%w for sc call event; num topics = %d", dto.ErrInvalidNumTopicsInEvent, len(topics))
 	}
 
 	return nil
 }
 
+// CheckValidity will receive the topics and validate them
+func (tc *topicsChecker) CheckValidity(topics [][]byte, transferData *sovereign.TransferData) error {
```

Comment:
Please delete this.
We should not code like this: `if check1 is passed => return no error`.
Each event processor should call its specific topic validity check function